packages/vue-remark/index.js

```diff
@@ -0,0 +1,165 @@
+
+    this.filesystem = new Filesystem(api, {
+      path: '**/*.md',
+      typeName: options.typeName,
+      baseDir: options.baseDir,
+      pathPrefix: options.pathPrefix,
+      route: options.route,
+      index: options.index,
+      refs: options.refs
+    })
+
+    api.store._addTransformer(RemarkTransformer, options.remark)
+
+    this.remark = api.store._transformers['text/markdown']
+
+    this.processor = this.remark.createProcessor({
+      plugins: [sfcSyntax, toVueRemarkAst],
+      stringifier: toSFC
+    })
+
+    api.transpileDependencies([path.resolve(__dirname, 'src')])
+    api.chainWebpack(config => this.chainWebpack(config))
+    api.createPages(args => this.createPages(args))
+
+    api.registerComponentParser({
+      test: /\.md/,
+      parse: this.parseComponent.bind(this)
+    })
+
+    api.___onCreateContentType(options => {
+      if (options.typeName === this.options.typeName) {
+        options.component = false
+      }
+    })
+  }
+
+  parseComponent (resourcePath) {
+    const pageQueryRE = /<page-query>([^</]+)<\/page-query>/
+    const source = fs.readFileSync(resourcePath, 'utf-8')
+    const ast = this.processor.parse(source)
+
+    return {
+      pageQuery: ast.children
+        .filter(node => node.type === 'html' && /^<page-query/.test(node.value))
+        .map(node => pageQueryRE.exec(node.value)[1])
+        .pop()
+    }
+  }
+
+  chainWebpack (config) {
+    const vueLoader = config.module.rule('vue').use('vue-loader')
+    const includePaths = this.options.includePaths.map(p => this.api.resolve(p))
+
+    config.module.rule('vue-remark')
+      .test(/\.md$/)
+      .include
+      .add(resourcePath => {
+        if (includePaths.some(path => resourcePath.startsWith(path))) {
+          return true
+        }
+
+        return resourcePath.startsWith(this.context)
+      })
+      .end()
+      .use('vue-loader')
+      .loader('vue-loader')
+      .options(vueLoader.toConfig().options)
+      .end()
+      .use('vue-remark-loader')
+      .loader(require.resolve('./lib/loader.js'))
+      .options(this)
+  }
+
+  async createPages ({ getContentType, createPage }) {
+    const contentType = getContentType(this.options.typeName)
+
+    contentType.collection.find().forEach(node => {
+      if (this.component) {
+        createPage({
+          path: node.path,
+          component: this.component,
+          queryVariables: node,
+          _meta: {
+            vueRemark: {
+              type: 'import',
+              path: node.internal.origin
+            }
+          }
+        })
+      } else {
+        createPage({
+          path: node.path,
+          component: node.internal.origin,
+          queryVariables: node
+        })
+      }
+    })
+  }
+
+  async parse (source, resourcePath = null) {
+    const { content, ...data } = this.remark.parse(source.trim())
+    const defaultLayout = require.resolve('./src/VueRemarkLayout.js')
```

Comment:
It'd be nice if the default layout could be specified in the options.

Reply:
The plugin had a `layout` option where you could set a default layout component. But we replaced it with a `component` option instead. They work very differently because the `component` option will be used as the route component and not just wrap the template.

We needed a way to use a component as the route component in order to have a shared `page-query` etc because the layout component could just have a `static-query`. You can read more about the `component` option [here](https://github.com/gridsome/gridsome/blob/75c9ccfa0fd865c205989b0e8fc4ea1777b05be6/packages/vue-remark/README.md#component). Hope it makes sense :)